### PR TITLE
Fix spc doesn't have metal attribute error

### PR DESCRIPTION
**Pull request by hwpang**
### Motivation or Problem
Fix spc doesn't have metal attribute error

### Description of Changes
Changed `r.metal` to `metal`. Also changed dummy variable `r` to `rev_react` for a meaningful name & consistency.

### Testing
Species doesn't have metal attribute error is gone after the changes

### CODE REVIEW

File: rmgpy/data/kinetics/family.py [hunk 1]
```diff
@@ -4239,14 +4239,14 @@ def fix_labels_mol(mol, root_labels):
                 rrev.is_forward = False
 
                 if estimate_thermo:
-                    for r in rrev.reactants:
-                        if r.thermo is None:
-                            therm_spc = deepcopy(r)
+                    for rev_react in rrev.reactants:
+                        if rev_react.thermo is None:
+                            therm_spc = deepcopy(rev_react)
```
Comment (mjohnson541) [inline rmgpy/data/kinetics/family.py:4244]: Can we avoid this deepcopy the same way we did in #2153 ?

**Review by mjohnson541 — CHANGES_REQUESTED** [review]
Looks good! One thing.

File: rmgpy/data/kinetics/family.py (hunk 1 above)
Comment (hwpang) [reply rmgpy/data/kinetics/family.py:4244]: Yeah sure!

**Comment by hwpang** [comment]
@mjohnson541 Thanks for the review! I have added a function `get_react_thermo` to avoid deepcopy.

File: rmgpy/data/kinetics/family.py [hunk 2]
```diff
@@ -4049,6 +4049,23 @@ def fix_labels_mol(mol, root_labels):
                 if atm.label not in root_labels:
                     atm.label = ''
 
+        def get_react_thermo(react,rxn_react,metal):
```
Comment (rwest) [inline rmgpy/data/kinetics/family.py:4052]: From the function name and parameter names I can't guess what this is supposed to do. The meanings of `react`, `rxn_react`, and `metal` are hard to guess out of context. Maybe consider renaming them, and/or add a docstring to the function saying what it does?

**Review by rwest — COMMENTED** [review]
I know you were just reducing code duplication by making that function, but there's an opportunity to make it better. As it stands, I found the function quite confusing based just on its name and parameter names.

**Comment by davidfarinajr** [comment]
Thanks for fixing this @hwpang !  I like adding the function here, but perhaps it could have just 2 arguments (`react` and `metal`) and then, since it's a get function, it could return the thermo object instead of assigning it to rxn_react.thermo attr?  So something like 
```
 if rxns[i].reactants[j].thermo is None:
      rxns[i].reactants[j].thermo = get_react_thermo(react,metal)
```

**Comment by davidfarinajr** [comment]
Also a little bit of documentation could help, but since it's a local function inside a function, I don't think we will use it too much

File: rmgpy/data/kinetics/family.py (hunk 2 above)
Comment (hwpang) [reply rmgpy/data/kinetics/family.py:4052]: @rwest Good point! Thanks

**Comment by hwpang** [comment]
@rwest @davidfarinajr Thanks for the comments! I changed the function name to `get_reactant_thermo` and added a docstring to give more explanation. I also changed the function to return the thermo object!

**Review by rwest — APPROVED** [review]

**Review by davidfarinajr — APPROVED** [review]
🙏 Thank you for fixing this @hwpang and addressing the comments !  Looks good to me!

**Comment by mjohnson541** [comment]
@hwpang can you rebase?

**Comment by hwpang** [comment]
> @hwpang can you rebase?

Rebased!

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**